synapse/lib/ast.py

```diff
@@ -1267,6 +1267,9 @@ async def run(self, runt, genr):
             count += 1
 
             item = s_stormtypes.fromprim(await self.kids[0].compute(runt, path), basetypes=False)
+            if item is None:
```

Comment:
`$foo=bar $foo.bar=baz` can also throw an `AttributeError`. Should we just try/catch around the `setitem()` calls ?

Reply:
Def worth considering str/int/bool/null values here. :thinking:

Reply:
It's not the call to `setitem` that's the problem. The problem is that the returned type doesn't have a `setitem` function because it's not a storm type. I updated this check to be anything that doesn't have a `setitem` property. I think that addresses your concern?

Reply:
Reverted this bug fix and opened SYN-6531 to address it later.